### PR TITLE
Update date.js

### PR DESCRIPTION
**Pull request by ArtiDjeims**
### Description
Date format doesn't make sense (AM/PM).

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14200) <br/>All committers have signed the CLA.

**Comment by sidolov** [comment]
Hi @ArtiDjeims , please, sign CLA, otherwise, I can't process you pull request, thank you!

**Comment by magento-engcom-team** [comment]
@ArtiDjeims thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by ArtiDjeims** [comment]
Done 💯

**Review by VladimirZaets — APPROVED** [review]

File: app/code/Magento/Ui/view/base/web/js/grid/columns/date.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ define([
 
     return Column.extend({
         defaults: {
-            dateFormat: 'MMM d, YYYY h:mm:ss A'
+            dateFormat: 'MMM d, YYYY h:mm:ss'
```
Comment (VladimirZaets) [inline app/code/Magento/Ui/view/base/web/js/grid/columns/date.js:18]: @ArtiDjeims In the current case, we use h:mm:ss format, in this the case 'h' means that we use 12h time format, so we should use A to show PM/AM abbreviation. 
You can look at the documentation of monent.js library that we use to work with a date and time.
https://momentjs.com/docs/#the-rfc-2822-date-time-format

**Review by VladimirZaets — CHANGES_REQUESTED** [review]

**Pull request closed without merging**